hibernate-core/src/main/java/org/hibernate/tool/schema/internal/AbstractSchemaMigrator.java

```diff
@@ -439,11 +443,38 @@ protected void applyForeignKeys(
 		}
 	}
 
-	private ForeignKeyInformation findMatchingForeignKey(ForeignKey foreignKey, TableInformation tableInformation) {
-		if ( foreignKey.getName() == null ) {
-			return null;
+	/**
+	 * Check if the ForeignKey already exists. First check based on definition and if that is not matched check if a key with the exact 
+	 * same name exists. Keys with the same name are presumed to be functional equal.
+	 * 
+	 * @param foreignKey - ForeignKey, new key to be created
+	 * @param tableInformation - TableInformation, information of existing keys
+	 * @return boolean, true if key already exists
+	 */
+	private boolean checkForExistingForeignKey(ForeignKey foreignKey, TableInformation tableInformation) {
```

Comment:
Hello Naros, do I need to change a status flag somewhere or is everybody just very busy. No problem but I just want to make sure I'm not missing something. Also the "Requires Test Case" test flag should be updated (I think...) but I can't do that.

Thanks,
Milo